ml-algorithms/src/main/java/org/opensearch/ml/engine/memory/MLMemoryManager.java

```diff
@@ -0,0 +1,164 @@
+import org.opensearch.ml.memory.action.conversation.GetInteractionsAction;
+import org.opensearch.ml.memory.action.conversation.GetInteractionsRequest;
+import org.opensearch.ml.memory.action.conversation.GetInteractionsResponse;
+import org.opensearch.ml.memory.action.conversation.GetTracesAction;
+import org.opensearch.ml.memory.action.conversation.GetTracesRequest;
+import org.opensearch.ml.memory.action.conversation.GetTracesResponse;
+import org.opensearch.ml.memory.action.conversation.UpdateInteractionAction;
+import org.opensearch.ml.memory.action.conversation.UpdateInteractionRequest;
+
+import com.google.common.base.Preconditions;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * Memory manager for Memories. It contains ML memory related operations like create, read interactions etc.
+ */
+@Log4j2
+@AllArgsConstructor
+public class MLMemoryManager {
+
+    private Client client;
+
+    /**
+     * Create a new Conversation
+     * @param name the name of the conversation
+     * @param applicationType the application type that creates this conversation
+     * @param actionListener action listener to process the response
+     */
+    public void createConversation(String name, String applicationType, ActionListener<CreateConversationResponse> actionListener) {
+        try {
+            client.execute(CreateConversationAction.INSTANCE, new CreateConversationRequest(name, applicationType), actionListener);
+        } catch (RuntimeException exception) {
+            actionListener.onFailure(exception);
+        }
+    }
+
+    /**
+     * Adds an interaction to the conversation indicated, updating the conversational metadata
+     * @param conversationId the conversation to add the interaction to
+     * @param input the human input for the interaction
+     * @param promptTemplate the prompt template used for this interaction
+     * @param response the Gen AI response for this interaction
+     * @param origin the name of the GenAI agent in this interaction
+     * @param additionalInfo additional information used in constructing the LLM prompt
+     * @param parentIntId the parent interactionId of this interaction
+     * @param traceNum the trace number for a parent interaction
+     * @param actionListener gets the ID of the new interaction
+     */
+    public void createInteraction(
+        String conversationId,
+        String input,
+        String promptTemplate,
+        String response,
+        String origin,
+        Map<String, String> additionalInfo,
+        String parentIntId,
+        Integer traceNum,
+        ActionListener<CreateInteractionResponse> actionListener
+    ) {
+        Preconditions.checkNotNull(conversationId);
+        Preconditions.checkNotNull(input);
+        Preconditions.checkNotNull(response);
+        // additionalInfo cannot be null as flat object
+        additionalInfo = (additionalInfo == null) ? new HashMap<>() : additionalInfo;
```

Comment:
Why need to create a new HashMap if `additionalInfo` is null?

Reply:
I think that was due to a bug fix. The flatobject cannot be null otherwise the index will fail. There was error reported before this line was added.